projects/rero/ng-core/src/lib/record/editor/type/password-generator-type.component.ts

```diff
@@ -0,0 +1,173 @@
+import { tap } from 'rxjs/operators';
+
+/**
+ * Available options for generateOptions (GeneratePassword):
+ * - length: Integer, length of password.
+ *   default: 10
+ * - lowercase: Boolean, put lowercase letters in password.
+ *   default: true
+ * - uppercase: Boolean, put uppercase letters in password.
+ *   default: true
+ * - numbers: Boolean, put numbers in password.
+ *   default: true
+ * - symbols: Boolean, put symbols in password.
+ *   default: false
+ * - exclude: String, characters to be excluded from password.
+ *   default: ''
+ * - minLengthLowercase: only if lowercase is set to true, minLengthLowercase will create
+ *   a password that will have minimum number of lower case characters in the password.
+ *   default: 1
+ * - minLengthUppercase: only if uppercase is set to true, minLengthUppercase will create
+ *   a password that will have minimum number of upper case characters in the password.
+ *   default: 1
+ * - minLengthNumbers: only if numbers is set to true, minLengthNumbers will create
+ *   a password that will have minimum number of numbers in the password.
+ *   default: 1
+ * - minLengthSymbols: only if symbols is set to true, minLengthSymbols will create
+ *   a password that will have minimum number of symbols in the password.
+ *   default: 1
+ */
+@Component({
+  selector: 'ng-core-editor-field-password-generator',
+  template: `
+  <div>
+    <div class="input-group mb-2 mr-sm-2">
+      <input
+        id="password"
+        [type]="type"
+        class="form-control"
+        autocomplete="off"
+        [formControl]="formControl"
+        (change)="onChange($event.target.value)"
+        [readonly]="to.readonly"
+      >
+      <div class="input-group-prepend">
+        <div class="input-group-text" (click)="onClick()">
+          <i class="fa fa-repeat" title="{{ 'Generate a new password' | translate }}"></i>
+        </div>
+        <div class="input-group-text" (click)="showHidePassword()">
+          <i class="fa" [ngClass]="{'fa-eye': type === 'password', 'fa-eye-slash': type === 'text'}" title="{{ 'Show or hide password' | translate }}"></i>
+        </div>
+        <div *ngIf="to.enabledEditMode" class="input-group-text" (click)="to.readonly = !to.readonly">
+          <i class="fa" [ngClass]="{'fa-lock': to.readonly, 'fa-unlock-alt': !to.readonly}" title="{{ 'Edit mode' | translate }}"></i>
+        </div>
+      </div>
+    </div>
+    <small class="form-text text-muted" *ngIf="!showError && generate" translate>
+      The password has been copied to the clipboard.
+    </small>
+  </div>
+  `
+})
+export class PasswordGeneratorTypeComponent extends FieldType implements OnInit {
+  /** Default options */
+  defaultOptions: Partial<FormlyFieldConfig> = {
+    templateOptions: {
+      enabledEditMode: false,
+      minLength: 8,
+      maxLength: 255,
+      specialChar: false,
+      api: undefined,
+      readonly: true,
+      generateOptions: {}
+    }
+  };
+
+  generate: boolean = false;
```

Comment:
It seems to means that the password has been generated, thus I would suggest to use `generated` or `hasBeenGenerated`.